### DISCUSSION TITLE
Consider warning about misuse of query parameters with BaseModel

### DISCUSSION THREAD

**Discussion by nvgoldin** [fastapi/fastapi#9089] [Questions]
I've noticed several times, especially for new users - a common mistake is to put for a in a GET request with query parameters, a `BaseModel` type annotation, causing the OpenAPI docs to miss the parameter entirely.
Consider the following code:

```Python
from pydantic import BaseModel
from fastapi import FastAPI


class UserDetails(BaseModel):
    username: str
    full_name: str


api = FastAPI()


@api.get('/user-details')
def get_user_details(details: UserDetails):
    pass
```
The rendered OpenAPI documentation parameters is empty:
![image](https://user-images.githubusercontent.com/13018704/87077867-6eb61e00-c22c-11ea-91ec-840fc8711d09.png)

Is this something that would be worth solving? I think it would ease the on-boarding of new users. 
I can think of two solutions:

1. Printing a warning message in debug mode when that usage is detected. 
2. Throwing an exception? probably too intrusive. 

What do yo think? would be happy to implement the solution.

**Comment by YuriiMotov** [comment]
In current version Swagger shows it as Body parameter, and if you try sending a request, it will show quite descriptive error message:

![image](https://github.com/user-attachments/assets/f0a6e764-b20a-4b1f-ab1a-f25142939c2b)
